cloudconfig/userdatacfg_unix.go

```diff
@@ -610,7 +612,37 @@ func (w *unixConfigure) setUpGUI() (func(), error) {
 		// so it has a chance to add it to its catalogue.
 		w.conf.AddRunCmd("rm -f $gui/gui.tar.bz2 $gui/jujugui.sha256 $gui/downloaded-gui.txt")
 	}, nil
+}
 
+// writeCleanupScript writes a shell script to /sbin that can be used to remove
+// all Juju services from a machine.
+// Once this script is run, logic to check whether such a machine is already
+// provisioned should return false and the machine can be reused as a target
+// for either bootstrap or add-machine.
+func (w *unixConfigure) writeCleanupScript() {
+	script := `
+#!/bin/bash
+
+# WARNING
+# This script will clean a host previously used to run a Juju controller/machine.
+# Running this on a live installation will render Juju inoperable.
+
+for path_to_unit in $(ls /etc/systemd/system/juju*); do
+  echo "removing juju service: $path_to_unit"
+  unit=$(basename "$path_to_unit")
+  systemctl stop "$unit"
+  systemctl disable "$unit"
+  systemctl daemon-reload
+done
+
+echo "removing /var/lib/juju/db/*"
+rm -rf /var/lib/juju/db/*
+
+echo "removing /var/lib/juju/raft/*"
+rm -rf /var/lib/juju/raft/*
+`[1:]
```

Comment:
Surely this can be a const, then you can just refer to it in the test. I don't see a need to verify that the output is the same between running it and testing it. You're always going to just accept the new changes you make to the script.

Reply:
I made it a const and wild-carded the test. The tests are in an external package and I don't want to export it just for that.